app/services/rent_service.rb

```diff
@@ -29,6 +31,9 @@ def self.update(params, user, rent)
     begin
       if rent.owner?(user)
         rent.update!(params)
+        rent.status = comfortable_hours?(rent.d_from, rent.d_til) ? :confirmed : :unconfirmed
+        RentMailer.rent_email(rent).deliver_now
+        rent.update!(params)
```

Comment:
Here you have
1. Update
2. Set status
3. Send mail
4. Update

This should be
1. Set status
2. Update
3. Send mail

Always persist changes before sending e-mails/notifications. Also avoid doing unnecessary read/writes to the DB if possible